### PR TITLE
Bring go implementation up to date

### LINKED ISSUE #55: Go

### PR DESCRIPTION
**Pull request by umgefahren**
I implemented the latest spec adjustments in Go. The code even cleans up behind itself.

Closes #55

### CODE REVIEW

File: go/bigstore.go [hunk 1]
```diff
@@ -0,0 +1,286 @@
...
+	"sync"
+	"sync/atomic"
+	"syscall"
+)
+
+type BigDataStore struct {
+	rootDir       string
+	fileIndex     map[string]string
+	fileIndexLock sync.Mutex
+	// fileHash        sync.Map
```
Comment (vypxl) [inline go/bigstore.go:25]: remove comment pls

File: go/bigstore.go [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+	globalContext, globalCancel := context.WithCancel(context.TODO())
+
+	sigs := make(chan os.Signal, 1)
+	signal.Notify(sigs, syscall.SIGTERM, syscall.SIGSEGV, syscall.SIGINT)
+
+	ret := new(BigDataStore)
+	ret.rootDir = rootDir
+	ret.fileIndex = make(map[string]string)
+	ret.fileIndexLock = sync.Mutex{}
+	// ret.fileHash = sync.Map{}
```
Comment (vypxl) [inline go/bigstore.go:60]: remove comment pls

File: go/bigstore.go [hunk 3]
```diff
@@ -0,0 +1,286 @@
...
+	b.fileIndexLock.Unlock()
+	_, err = conn.Write([]byte("READY\n"))
+	if err != nil {
+		return err
+	}
+	hashValue, err := b.CopySmart(conn, newFile, size)
+	if err != nil {
+		return err
+	}
+	// b.fileHash.Store(name, hashValue)
```
Comment (vypxl) [inline go/bigstore.go:115]: remove comment pls

File: go/server.go [hunk 4]
```diff
@@ -5,47 +5,47 @@ import (
...
-		        }
-		        connChan <- inChan
-
-                }
+		for {
+			connection, err := ln.Accept()
+			// fmt.Println("New Connection")
+			if errors.Is(err, net.ErrClosed) {
+				// fmt.Println("Connection is closed")
+				return
```
Comment (vypxl) [inline go/server.go:31]: remove comment pls

File: go/server.go [hunk 5]
```diff
@@ -5,47 +5,47 @@ import (
...
+		}
 	}()
 	for {
 		select {
-		case nConn := <- connChan:
+		case nConn := <-connChan:
 			if nConn.connerr != nil {
 				return nConn.connerr
 			}
 			// fmt.Println("Spawning new handler")
```
Comment (vypxl) [inline go/server.go:46]: remove comment pls

**Review by vypxl — COMMENTED** [review]
remove comments pls then good

File: go/bigstore.go (hunk 1 above)
Comment (umgefahren) [reply go/bigstore.go]: Removed

File: go/bigstore.go (hunk 2 above)
Comment (umgefahren) [reply go/bigstore.go:59]: Removed

File: go/bigstore.go (hunk 3 above)
Comment (umgefahren) [reply go/bigstore.go]: Remonde

**Comment by umgefahren** [comment]
I have removed all of your complaints

**Comment by vypxl** [comment]
Thanks

**Review by vypxl — APPROVED** [review]

**Pull request merged**